src/processors/AvroSchemaInputProcessor.ts

```diff
@@ -0,0 +1,152 @@
+import { AbstractInputProcessor } from './AbstractInputProcessor';
+import {
+  CommonModel,
+  InputMetaModel,
+  AvroSchema,
+  ProcessorOptions
+} from '../models';
+import { Logger } from '../utils';
+import { Interpreter } from '../interpreter/Interpreter';
+import { convertToMetaModel } from '../helpers';
+
+/**
+ * Class for processing Avro Schema input
+ */
+export class AvroSchemaInputProcessor extends AbstractInputProcessor {
+  /**
+   * Function processing an Avro Schema input
+   *
+   * @param input
+   */
+
+  shouldProcess(input?: any): boolean {
+    if (
+      input === '' ||
+      JSON.stringify(input) === '{}' ||
+      JSON.stringify(input) === '[]'
+    ) {
+      return false;
+    }
+    if (!input.type) {
+      return false;
+    }
+    return true;
+  }
+
+  process(input?: any, options?: ProcessorOptions): Promise<InputMetaModel> {
+    if (!this.shouldProcess(input)) {
+      return Promise.reject(
+        new Error('Input is not an Avro Schema, so it cannot be processed.')
+      );
+    }
+    Logger.debug('Processing input as Avro Schema document');
+    const inputModel = new InputMetaModel();
+    inputModel.originalInput = input;
+    input = AvroSchemaInputProcessor.reflectSchemaNames(
+      input,
+      {},
+      'root',
+      true
+    ) as any;
+    const parsedSchema = AvroSchema.toSchema(input);
+    const newCommonModel = AvroSchemaInputProcessor.convertSchemaToCommonModel(
+      parsedSchema,
+      options
+    );
+    const metaModel = convertToMetaModel(newCommonModel);
```

Comment:
You dont need to convert Avro to common model only convert it to meta model directly 🙂 

This means that you build the meta model from the Avro schema: https://modelina.org/examples?selectedExample=meta-model